# 船只轨迹跟踪的配置文件
# 配置说明：请根据实际设备参数修改以下配置

camera:
  video_path: "/home/tl/RV/src/marnav_vis/clip-01/2022_06_04_12_05_12_12_07_02_b.mp4"
  camera_start_timestamp: 1654315512000  # 毫秒级时间戳
  camera_publish_fps: 25
  width_height: [2560, 1440]  # 注意這個數值應爲輸入視頻的原始尺寸，因爲camera_intrinsic_para是根據這個尺寸設置的
  noise_range_ns: 10000000  # 10毫秒的纳秒表示
  angle_between_cameras: 0 # 相机之间的角度差
  camera_parameters:  # 相机配置列表（支持动态增减相机数量）
    - camera_name: "camera_0"           # 相机名称（用于映射）
      topic_name: "/camera_image_topic_0"   # 相机0的话题名
      camera_index: 0                       # 相机索引
      camera_matrix:   # 数据集模式，不需要获取实际的相机内参，用数据集里的相机内参初始化
        fov_hor: 55.0
        fov_ver: 30.94
        fx: 2391.26
        fy: 2446.89
        u0: 1305.04
        v0: 855.214
    - camera_name: "camera_1"           # 相机名称（用于映射）
      topic_name: "/camera_image_topic_1"   # 相机1的话题名
      camera_index: 1                       # 相机索引
      camera_matrix:
        fov_hor: 55.0
        fov_ver: 30.94
        fx: 2391.26
        fy: 2446.89
        u0: 1305.04
        v0: 855.214
    - camera_name: "camera_2"           # 相机名称（用于映射）
      topic_name: "/camera_image_topic_2"   # 相机2的话题名 camera_image_topic_2
      camera_index: 2                       # 相机索引
      camera_matrix:
        fov_hor: 55.0
        fov_ver: 30.94
        fx: 2391.26
        fy: 2446.89
        u0: 1305.04
        v0: 855.214


gnss:
  gnss_publish_rate: 5.0
  gnss_pub_topic: "/gnss_topic"
  camera_gnss_para:
    lon: 114.32583
    lat: 30.60139
    horizontal_orientation: 352.0
    vertical_orientation: -4.0
    camera_height: 20.0

ais:
  ais_csv_folder: "/home/tl/RV/src/marnav_vis/clip-01/ais/"
  ais_start_timestamp: 1654315512000
  ais_csv_topic: "/ais_csv_topic"
  ais_batch_pub_topic: "/ais_batch_topic_offline"

DeepSORVF:
  fus_trajectory_topic: "/fus_trajectory_topic"
  get_camera_params_service: "/get_camera_params_service"
  input_fps: 25 # 进程刷新频率
  output_fps: 10 # 当前跟踪结果的显示频率
  anti: 1 # 是否进行抗遮挡处理
  anti_rate: 0 # 启用抗遮挡处理的阈值
  sync_queue_size: 10 # 同步队列大小
  sync_slop: 0.1 # 同步时间误差
  skip_interval: 1000 # 单位ms，表示在各个相机对应的跟踪进程中，每隔skip_interval ms处理一帧
  camera_type: "normal" # 选择相机类型: normal, fisheye, 
